backend/utilities/auth.py

```
@@ -85,31 +88,38 @@ async def get_current_user(
         raise credentials_exception
     
     user_id_str: str = payload.get("sub")
     if user_id_str is None:
+        logger.warning("Authentication failed: No user ID in token payload")
         raise credentials_exception
     
     try:
         user_id = int(user_id_str)
     except (ValueError, TypeError):
+        logger.warning("Authentication failed: Invalid user ID format")
         raise credentials_exception
```

Ruff (0.14.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)